File: data/compilers_comparison/src/examples/second_example/src/func36.c
#include "second_example.h" 
array_t* func36(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array97 = func37(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array97->id);
      free(params0.data);
      array97->refC--;
      if(array97->refC == 0) {
         free(array97->data);
         free(array97);
         DEBUG_FREE(array97->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array106 = func38(&params0, rng(), loopsFactor);
      DEBUG_RETURN(array106->id);
      free(params0.data);
      array106->refC--;
      if(array106->refC == 0) {
         free(array106->data);
         free(array106);
         DEBUG_FREE(array106->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array111 = func39(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array111->id);
   free(params0.data);
   return array111;
}

File: data/multilang_comparison/program_julia/src/func36.jl
include("program_julia_head.jl")
function func36(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array49 = func37(params0, loopsFactor);
      array49.refC -= 1
      if array49.refC == 0
          empty!(array49.data)
          array49 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array51 = func38(params0, get_path(), loopsFactor);
      array51.refC -= 1
      if array51.refC == 0
          empty!(array51.data)
          array51 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array55 = func39(params0, loopsFactor);
   return array55;
end

